source/client/process_bootstrap.cc

```diff
@@ -0,0 +1,123 @@
+#include "source/client/process_bootstrap.h"
+
+#include <string>
+#include <vector>
+
+#include "nighthawk/client/options.h"
+#include "nighthawk/common/uri.h"
+
+#include "external/envoy/source/common/common/statusor.h"
+#include "external/envoy_api/envoy/config/bootstrap/v3/bootstrap.pb.h"
+
+#include "source/client/sni_utility.h"
+
+namespace Nighthawk {
+namespace {
+
+using envoy::config::bootstrap::v3::Bootstrap;
+
+// Adds a cluster for the request source into the bootstrap.
+void addRequestSourceCluster(const Client::Options& options, const Uri& uri, int worker_number,
+                             Bootstrap& bootstrap) {
+  auto* cluster = bootstrap.mutable_static_resources()->add_clusters();
+  cluster->mutable_http2_protocol_options();
+  cluster->set_name(fmt::format("{}.requestsource", worker_number));
+  cluster->set_type(
+      envoy::config::cluster::v3::Cluster::DiscoveryType::Cluster_DiscoveryType_STATIC);
+  cluster->mutable_connect_timeout()->set_seconds(options.timeout().count());
+
+  auto* load_assignment = cluster->mutable_load_assignment();
+  load_assignment->set_cluster_name(cluster->name());
+  auto* socket = cluster->mutable_load_assignment()
+                     ->add_endpoints()
+                     ->add_lb_endpoints()
+                     ->mutable_endpoint()
+                     ->mutable_address()
+                     ->mutable_socket_address();
+  socket->set_address(uri.address()->ip()->addressAsString());
+  socket->set_port_value(uri.port());
+}
+
+} // namespace
+
+absl::StatusOr<Bootstrap> createBootstrapConfiguration(const Client::Options& options,
+                                                       const std::vector<UriPtr>& uris,
+                                                       const UriPtr& request_source_uri,
+                                                       int number_of_workers) {
+  Bootstrap bootstrap;
+  for (int i = 0; i < number_of_workers; i++) {
+    auto* cluster = bootstrap.mutable_static_resources()->add_clusters();
+
+    if (uris.empty()) {
```

Comment:
preexisting - why is this within the for loop?  given that we're basically asserting a parameter's existence, shouldn't this be the first thing this function does?

Reply:
Good point, will address this in the next PR to avoid merge conflicts.

https://github.com/envoyproxy/nighthawk/pull/721

Reply:
Done in bfd032b6adbd820450ea99cf7edbd05f0130cc84.